backend/lib/spree/backend_configuration.rb

```diff
@@ -6,6 +6,16 @@ module Spree
   class BackendConfiguration < Preferences::Configuration
     preference :locale, :string, default: I18n.default_locale
 
+    preference :frontend_product_path,
+      :proc,
+      default: proc {
+        ->(template_context, product) {
+          return unless template_context.spree.respond_to?(:product_path)
+
+          template_context.spree.product_path(product)
+        }
```

Comment:
@waiting-for-dev Appreciate to head your thoughts on this fixup. Prior to this PR, the behavior of the backend product link is as follows:

1. If the app has a frontend component, the backend would display the link. For instance, this is the behavior of the Solidus sandbox app.
2. If the app does not have a frontend component, the backend would NOT display the product link. This is how the backend feature spec suite behaves, since it does not have an integrated frontend.

I've attempted to maintain this behavior in this default value for the `frontend_product_path` preference. What do you think? On the upside, it doesn't introduce changes to the feature. However, on the downside, it seems to look a bit complicated for a default value.

The good news is that if an app has to override this `frontend_product_path`, it does not necessarily have to match the behavior of this default value. In particular, for a Starter FE-based app, the app already knows that it has a frontend, so it can skip the `respond_to?(:product_path)` guard:

```rb
# config/initializers/spree.rb
Spree::Backend::Config.configure do |config|
  config.frontend_product_path = ->(template_context, product) { template_context.spree.product_path(product) }
end
```

If you feel that the default value for `frontend_product_path` is too complicated, I think the next best alternative is to set the default value to nil, like this:

```rb
# backend/lib/spree/backend_configuration.rb
preference :frontend_product_path, :proc, default: nil
```

```erb
<!-- backend/app/views/spree/admin/products/edit.html.erb  ->
  <% if Spree::Backend::Config[:frontend_product_path] %>
    <li id="view_product_link">
      <%= link_to t('spree.view_product'), Spree::Backend::Config[:frontend_product_path].call(self, @product), id: 'admin_view_product', class: 'btn btn-default ml-2' %>
    </li>
  <% end %>
```

I think this implementation is straightforward and it allows the backend feature spec suite to pass without any changes. The downside is that this is a breaking change: users would have to override this preference if they would like to continue showing the product link on their backend.

What do you think?

Reply:
Hey @gsmendoza, I see your point. I'd stick with the first one. You're right; it's complicated, but it's what it's giving us the behavior we need :slightly_smiling_face: Nothing guarantees us that the method `product_path` is present, so, yeah, we need to check it!

Reply:
@waiting-for-dev Cool! I'll go ahead with implementing the first option :+1: